### ISSUE TITLE
Request: Support 7-inch and 10-inch Android tablets

### ISSUE THREAD

**Issue by apaatsio** [mmcc007/screenshots#8]
The [Android guidelines](https://developer.android.com/docs/quality-guidelines/tablet-app-quality#google-play) recommend to *"Add screenshots taken on both 7-inch and 10-inch tablets."* to Google Play. I think these should be supported by `screens.yaml`.

**Comment by mmcc007** [comment]
Good point... I have a placeholder for that for now:
https://github.com/mmcc007/screenshots/blob/master/lib/resources/screens.yaml#L45

It should be just a matter of getting the art for the status and navigation bars, and the frame, and then updating the `screens.yaml`. ... and a minor code change.

So the blocker is locating or creating the art. (I haven't searched that much for the art so far...)

**Comment by mmcc007** [comment]
BTW: the artwork for the frames is here:
https://facebook.design/devices

Your input would be appreciated regarding any preference for particular android 7in/10in device (not sure which ones are preferred). Will probably use one of each size as a default to get started.

The other thing needed is to locate (or create) the artwork for the matching status bars and navigation bars. 

Sketch files for the nav and status bars are useful because the artwork is usually complete and accurate and can be exported as a png. 

Not sure if you can provide input on graphic art, but if you have any input on location of suitable sketch files that would be great!

Once the graphic art is complete, the png (of a status bar or a navigation bar) can be added as a resource in the screenshots package.

At this point, the new 7in/10in devices can be added to 
https://github.com/mmcc007/screenshots/blob/master/lib/resources/screens.yaml
test, and then released!

**Comment by apaatsio** [comment]
I don't have any preference for the models. I don't even know how such a preference should be established. Market share, brand, design, screen resolution...?

My best *guess* is that any models by popular brands - like Samsung, Amazon, Huawei, Lenovo - will be fine.

**Comment by mmcc007** [comment]
It's not clear to me which if the devices in https://facebook.design/devices are 7 or 10 inch android tablets. Can you identify any?

BTW: It just occurred to me whether Flutter supports android tablets?
https://flutter.dev/docs/resources/faq#what-devices-and-os-versions-does-flutter-run-on
It looks like they don't have a definitive list.

Do you know of any Flutter apps in the Play Store that run on android tablets?

If there are enough apps out there in the wild, then it makes sense to support them. Otherwise, it seems like there is no (or not enough) demand for support in Screenshots. What do you think?

**Comment by apaatsio** [comment]
Tablet is a lie. There is no such thing as a tablet. There are only Android devices. We just choose to call some devices smartphones and some devices tablets, some are even called phablets. What the exact definitions for those labels are, that I do not know.

Any Flutter app will work on both smartphones and tablets *unless* the developer explicitly blocks some device models or blocks devices by screen size, for example.

**Comment by mmcc007** [comment]
Seems like there is some minor confusion in the android marketplace regarding multiple device types, when, as you say, there may only in fact be one device type. 

There are ongoing discussions about what android device types to support in Flutter (for example, see https://github.com/flutter/flutter/issues/9253)

For this issue, the blocker is the need for the art (a collection of PNGs) for a 7 and 10 inch device (that is supported by Flutter).

Once the art blocker is resolved, adding to the code is pretty straight forward.

So far, I have not been able to locate the art, and am not setup to create it (especially the frames). So for now, I am standing by for some assistance in resolving this art blocker.

So, for now, makes sense to wait and see if there is enough interest (thumbs-up, etc..) to locate/create the art.

**Comment by apaatsio** [comment]
Issue flutter/flutter#9253 is not a phone-vs-tablet issue, it's an x86-vs-ARM issue. Apparently, there are some Android devices that have x86 processors that Flutter doesn't currently support. But these are such a small minority that they are not relevant here.

I found some Samsung device skins here: https://developer.samsung.com/galaxy/emulator-skin, but I didn't find any info about the license for those images, though.

**Comment by mmcc007** [comment]
I can't find the screen size specs for 7 and 10 inch screens anywhere so far (you would think the Play Store would have specs but not so far). But per your samsung link I have:

| Display      | Resolution |
| ----------- | ----------- |
| 7.0 inches   | 800 x 1280        |
| 10.1 inches      | 2560 x 1600       |

Is this correct for the Play Store?

If the skins in the link are at the correct resolution required by the Play Store, then might be able to use them as a starting point.

**Comment by mmcc007** [comment]
Ah, according to Android Studio:

| Display      | Resolution | Density|
| ----------- | ----------- |----|
| 7" WSVGA (Tablet)   | 600 x 1024        |mdpi|
| 10.1" WXGA (Tablet)      | 800 x 1280      |mdpi|

These are the specs for the some of the supported tablet emulators (AVDs), so should probably be looking for matching skins/frames (and status/nav bars).

**Comment by apaatsio** [comment]
The guidelines only say "JPEG or 24-bit PNG (no alpha). Min length for any side: 320px. Max length for any side: 3840px."

The point is that the user gets some idea of how the app will look like on their device. The exact resolution is irrelevant. I would rather go with a high-resolution image than not. A low-resolution image might look smudgy on a large tablet with high DPI.

**Comment by mmcc007** [comment]
Since haven't settled on frames (skins) yet, it might be easier to move forward if some code is added to skip framing on android tablets. Can track the missing frames as a separate issue. 

So, for now, just need the final specs on screen sizes. 

Can you make a recommendation?

| Display      | Resolution |
| ----------- | ----------- |
| 7.0 inches   | ???? x ????        |
| 10.1 inches      | ???? x ????       |

**Comment by Alecsplus** [comment]
Dear all, there is a prevision about the time for the availability of the solution for this issue? Can I help you in any way?

**Comment by mmcc007** [comment]
This is a purely graphics art problem. There should probably be no new code required. A solution is driven by the level of difficulty and the level of demand among screenshots users in finding a solution.

The problem is with sourcing the images for the skins/frames, the status bars, and the navigation bars. Skins/frames for other screen sizes were found at 
https://facebook.design/devices
The only image provided there for an android tablet is for the [Nexus 9](https://en.wikipedia.org/wiki/Nexus_9) (aspect ratio 4:3)

Status bars and navigation bars were found by searching the internet and some editing in [Sketch](https://www.sketch.com/)

The sketch files for all sourced images are located in:
https://github.com/mmcc007/screenshots/tree/master/assets
and the resulting images are located in:
https://github.com/mmcc007/screenshots/tree/master/lib/resources

Since the Nexus 9 skin/frame is already available, one possible way to proceed is to source the status bar and navigation bar for the Nexus 9 which has the aspect ratio of a typical android tablet. (Actual images would be enough, images with matching sketch files would be preferred in case changes are needed in future).

If images for the the status bar and navigation bar for the Nexus 9 can be sourced (or created), then we can proceed.

**Comment by mmcc007** [comment]
I was able to locate the status bar and navigation bar images for Nexus 9. I added the images and related sketch files, configured the screens.yaml and tested it.

You can test this by installing the Nexus 9 from Android Studio, updating your screenshots.yaml and running screenshots on the dev version of screenshots. 

To install the latest dev version of screenshots:
```
git clone https://github.com/mmcc007/screenshots.git
cd screenshots
git checkout dev
pub global activate --source path .
```

Let me know how you get along.

**Comment by Alecsplus** [comment]
Dear Maurice, I finally found some time to download and open the project... Excuse my ignorance but I have not understood how to compile it in order to try it.-- :(

Thanks for your reply

**Comment by mmcc007** [comment]
@Alecsplus There should be no need to compile it, as it compiles automatically during installation.

Here is what a screenshot on the Nexus 9 looks like from the example app (if you add the 'Nexus 9' in the `screenshots.yaml`):
![Nexus 9-1](https://user-images.githubusercontent.com/1980785/58357103-70593280-7e2e-11e9-9e73-ab11a1e37a2a.png)

You should be able to get something similar for your app.

Let me know how you get along.

**Comment by Alecsplus** [comment]
Dear Maurice, I finally test the new version. It's perfect for portrait orientation, still have some problem on landscape.
See  attachment.

![Nexus 9-Settings](https://user-images.githubusercontent.com/4600233/58421170-97fb0580-808f-11e9-82d4-68f13ef71477.png)

**Comment by Alecsplus** [comment]
Ok, i competed my tests.
Play store ask examples for 7'' and 10'' tablet.
The screenshot produced are ok for 10''. Only the folder name is not correct (to use fastlane): should be **tenInchScreenshots** and not tabletScreenshots

When 7'' will be available do not hesitate to contact me for testing phase

**Comment by mmcc007** [comment]
@Alecsplus Glad you got it working for portrait. 

I changed the name of the folder in dev. Give it a shot and let me know.

Currently there is no support for landscape. It is possible to add the code to do this. However, this would involve rotating the frame to fit. A rotated frame is not recommended (AFAIK) for upload to the stores. Also, would have to source (or create) the images for the status bar and the navigation bar for all the currently supported devices (and any devices added in the future). 

If you just want to check the layout in landscape, it is possible to add code to check for a screenshot in landscape mode and leave that image unprocessed (not suitable for upload to stores).

Landscape would be a new feature, so, if you want it, I recommend creating a new issue and describing your use case.

**Comment by mmcc007** [comment]
BTW: per #50 the format of screenshots.yaml has changed. Here is an example:
```
tests:
  - test_driver/app.dart

staging: /tmp/screenshots

locales:
  - en-AU

devices:
  ios:
    iPhone 7 Plus:
      frame: false
    iPad Pro (12.9-inch) (2nd generation):
      frame: false
  android:
    Nexus 9:
    Nexus 6P:

# Frame screenshots
frame: false
```

**Comment by mmcc007** [comment]
@Alecsplus It is possible for _screenshots_ to set the orientation to landscape on an android emulator using something like:
```
adb shell settings put system user_rotation 1
```
I have not yet found the equivalent for an iOS simulator (I assume it exists).

It is possible to handle landscape on an exception basis by doing no processing on the images and depositing them somewhere for review.

At a later point, we could start processing these images.

If interested in a feature like this, let me know by creating a new issue.
